File: client/lib/features/chat/domain/message.dart
class Message {
  final String id;
  final String conversationId;
  final String senderId;
  final String type;
  final String? content;
  final String? mediaUrl;
  final int? duration;
  final String status;
  final DateTime? createdAt;
  
  Message({
    required this.id,
    required this.conversationId,
    required this.senderId,
    required this.type,
    this.content,
    this.mediaUrl,
    this.duration,
    required this.status,
    this.createdAt,
  });
  
  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['id']?.toString() ?? '',
      conversationId: json['conversation_id']?.toString() ?? '',
      senderId: json['sender_id']?.toString() ?? '',
      type: json['type']?.toString() ?? 'text',
      content: json['content']?.toString(),
      mediaUrl: json['media_url']?.toString(),
      duration: json['duration'] is int ? json['duration'] : null,
      status: json['status']?.toString() ?? 'sent',
      createdAt: json['created_at'] != null ? DateTime.tryParse(json['created_at']) : null,
    );
  }
}

File: server/internal/model/message.go
package model

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
	"time"
)

type MessageType string
type MessageStatus string

const (
	MessageTypeText  MessageType = "text"
	MessageTypeImage MessageType = "image"
	MessageTypeVoice MessageType = "voice"
	MessageTypeVideo MessageType = "video"
	MessageTypeFile  MessageType = "file"
)

const (
	MessageStatusSending   MessageStatus = "sending"
	MessageStatusSent      MessageStatus = "sent"
	MessageStatusDelivered MessageStatus = "delivered"
	MessageStatusRead      MessageStatus = "read"
	MessageStatusFailed    MessageStatus = "failed"
)

type Message struct {
	ID             uuid.UUID     `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	ConversationID uuid.UUID     `gorm:"type:uuid;index;not null" json:"conversation_id"`
	SenderID       uuid.UUID     `gorm:"type:uuid;index;not null" json:"sender_id"`
	Type           MessageType   `gorm:"type:varchar(20);not null" json:"type"`
	Content        string        `gorm:"type:text" json:"content"`
	MediaURL       string        `gorm:"size:255" json:"media_url"`
	Duration       int           `gorm:"default:0" json:"duration"`
	Status         MessageStatus `gorm:"type:varchar(20);default:'sent'" json:"status"`
	CreatedAt      time.Time     `gorm:"index" json:"created_at"`
}

func (m *Message) BeforeCreate(tx *gorm.DB) error {
	if m.ID == uuid.Nil {
		m.ID = uuid.New()
	}
	return nil
}
